editor/src/clj/editor/gui.clj

```diff
@@ -661,11 +746,190 @@
+                override-node-ids)]
+
+    (if (coll/empty? node-properties)
+      {}
+      (let [node-type (g/node-type root-node)
+            stripped-prop-kws (node-type-deref->stripped-prop-kws @node-type)]
+        (persistent!
+          (reduce dissoc! node-properties stripped-prop-kws))))))
+
+(defn- update-layout-property [evaluation-context node-id prop-kw update-fn & args]
+  (let [old-value (g/node-value node-id prop-kw evaluation-context)
+        new-value (apply update-fn old-value args)
+        current-layout (g/node-value node-id :current-layout evaluation-context)]
+    (if (str/blank? current-layout)
+      (g/set-property node-id prop-kw new-value)
+      (g/update-property
+        node-id :layout->prop->override
+        update current-layout
+        assoc prop-kw new-value))))
+
+(defn layout-property-edit-type-set-impl [evaluation-context node-id prop-kw old-value new-value changes-fn]
+  (let [current-layout (g/node-value node-id :current-layout evaluation-context)
+        changes (if (nil? changes-fn)
+                  {prop-kw new-value}
+                  (changes-fn evaluation-context node-id prop-kw old-value new-value))]
+    (if (str/blank? current-layout)
+      (into []
+            (mapcat (fn [[prop-kw value]]
+                      (if (nil? value)
+                        (g/clear-property node-id prop-kw)
+                        (g/set-property node-id prop-kw value))))
+            changes)
+      (g/update-property
+        node-id :layout->prop->override
+        update current-layout
+        (fn [prop->override]
+          (reduce-kv (fn [prop->override prop-kw new-value]
+                       (if (nil? new-value)
+                         (dissoc prop->override prop-kw)
+                         (assoc prop->override prop-kw new-value)))
+                     prop->override
+                     changes))))))
+
+(defn basic-layout-property-edit-type-clear-fn [node-id prop-kw]
+  (let [current-layout (g/node-value node-id :current-layout)]
+    (if (str/blank? current-layout)
+      (g/clear-property node-id prop-kw)
+      (g/update-property node-id :layout->prop->override eutil/dissoc-in [current-layout prop-kw]))))
+
+(defn layout-property-edit-type-clear-impl [node-id prop-kw changes-fn]
+  (let [[current-layout cleared-prop-kws]
+        (g/with-auto-evaluation-context evaluation-context
+          (pair (g/node-value node-id :current-layout evaluation-context)
+                (keys (changes-fn evaluation-context node-id prop-kw nil nil))))]
+    (when (coll/not-empty cleared-prop-kws)
+      (if (str/blank? current-layout)
+        (coll/mapcat
+          #(g/clear-property node-id %)
+          cleared-prop-kws)
+        (g/update-property
+          node-id :layout->prop->override
+          update current-layout
+          (fn [prop->override]
+            (coll/not-empty
+              (apply dissoc prop->override cleared-prop-kws))))))))
```

Comment:
We use the `:edit-type` mechanism to ensure the property editor updates values in the correct place. Either directly on the edited property, or inside the `:layout->prop->override` map.